examples/migrating_trench/trench_example.py

```diff
@@ -0,0 +1,196 @@
+
+# set horizontal diffusivity parameter
+options.horizontal_diffusivity = Constant(diffusivity)
+options.horizontal_viscosity = Constant(viscosity_hydro)
+
+# crank-nicholson used to integrate in time system of ODEs resulting from application of galerkin FEM
+options.timestepper_type = 'CrankNicolson'
+options.timestepper_options.implicitness_theta = 1.0
+options.norm_smoother = Constant(0.1)
+
+if not hasattr(options.timestepper_options, 'use_automatic_timestep'):
+    options.timestep = dt
+
+# make sure set all hydrodynamic and sediment flags before creating model
+solver_obj.create_sediment_model(uv_init=uv, elev_init=elev)
+c = SedimentTotalMassConservation2DCallback('sediment_2d',
+                                            solver_obj, export_to_hdf5=True, append_to_log=False)
+solver_obj.add_callback(c, eval_interval='timestep')
+
+# set boundary conditions
+
+left_bnd_id = 1
+right_bnd_id = 2
+
+swe_bnd = {}
+
+swe_bnd[left_bnd_id] = {'flux': Constant(-0.22)}
+swe_bnd[right_bnd_id] = {'elev': Constant(0.397)}
+
+solver_obj.bnd_functions['shallow_water'] = swe_bnd
+
+if options.sediment_model_options.solve_suspended_sediment:
+    solver_obj.bnd_functions['sediment'] = {
+        left_bnd_id: {'flux': Constant(-0.22), 'equilibrium': None},
+        right_bnd_id: {'elev': Constant(0.397)}}
+
+    # set initial conditions
+    solver_obj.assign_initial_conditions(uv=uv, elev=elev, sediment=solver_obj.sediment_model.get_equilibrium_tracer())
+
+else:
+    # set initial conditions
+    solver_obj.assign_initial_conditions(uv=uv, elev=elev)
+
+# run model
+solver_obj.iterate(export_func=export_bath_func)
+
+# record final sediment and final bathymetry
+xaxisthetis1 = []
+sedimentthetis1 = []
+baththetis1 = []
+
+for i in np.linspace(0, 15.8, 80):
+    xaxisthetis1.append(i)
+    if conservative:
+        d = solver_obj.fields.bathymetry_2d.at([i, 0.55]) + solver_obj.fields.elev_2d.at([i, 0.55])
+        sedimentthetis1.append(solver_obj.fields.sediment_2d.at([i, 0.55])/d)
+        baththetis1.append(solver_obj.fields.bathymetry_2d.at([i, 0.55]))
+    else:
+        sedimentthetis1.append(solver_obj.fields.sediment_2d.at([i, 0.55]))
+        baththetis1.append(solver_obj.fields.bathymetry_2d.at([i, 0.55]))
+
+# check sediment conservation
+sediment_mass_int, sediment_mass_int_rerr = solver_obj.callbacks['timestep']['sediment_2d total mass']()
+print("Sediment total mass error: %11.4e" % (sediment_mass_int_rerr))
+
+# check sediment and bathymetry values using previous runs
+sediment_solution = pd.read_csv('sediment.csv')
+bed_solution = pd.read_csv('bed.csv')
+
+
+assert max([abs((sediment_solution['Sediment'][i] - sedimentthetis1[i])/sediment_solution['Sediment'][i]) for i in range(len(sedimentthetis1))]) < 0.15, "error in sediment"
+
+assert max([abs((bed_solution['Bathymetry'][i] - baththetis1[i])) for i in range(len(baththetis1))]) < 0.01, "error in bed level"
```

Comment:
For this whole section I'm wondering whether that's the most appropriate in a user example: this is all done in the test already as well, no? comparing with a previous run is dangerous as this is now no longer run as a test, so this example will likely break at some point in the future without us noticing it... so rather than tests and checks I would prefer something more fluffy and "attractive" for a user

Reply:
I have removed this section and replaced it with a plot comparing model results to experimental data